plugins/providers/hyperv/cap/configure_disks.rb

```diff
@@ -0,0 +1,235 @@
+        def self.handle_configure_disk(machine, disk, all_disks)
+          disk_metadata = {}
+
+          # Grab the existing configured disk, if it exists
+          current_disk = get_current_disk(machine, disk, all_disks)
+
+          # Configure current disk
+          if !current_disk
+            # create new disk and attach
+            disk_metadata = create_disk(machine, disk)
+          elsif compare_disk_size(machine, disk, current_disk)
+            disk_metadata = resize_disk(machine, disk, current_disk)
+          else
+            # TODO OLD: What if it needs to be resized?
+
+            # Can't use "attached" attribute, because it is false when the guest
+            # is powered off
+            #
+            #disk_info = machine.provider.driver.get_disk(current_disk["Path"])
+            #if disk_info["Attached"] == false
+            #  LOGGER.warn("Disk '#{disk.name}' is not connected to guest '#{machine.name}', Vagrant will attempt to connect disk to guest")
+            #  machine.provider.driver.attach_disk(nil, nil, nil, current_disk["Path"])
+            #else
+            #  LOGGER.info("No further configuration required for disk '#{disk.name}'")
+            #end
+
+            disk_metadata = {UUID: current_disk["DiskIdentifier"], Name: disk.name, Path: current_disk["Path"]}
+            if disk.primary
+              disk_metadata[:primary] = true
+            end
+          end
+
+          disk_metadata
+        end
+
+        # Check to see if current disk is configured based on defined_disks
+        #
+        # @param [Kernel_V2::VagrantConfigDisk] disk_config
+        # @param [Hash] defined_disk
+        # @return [Boolean]
+        def self.compare_disk_size(machine, disk_config, defined_disk)
+          # Hyper-V returns disk size in bytes
+          requested_disk_size = disk_config.size
+          disk_actual = machine.provider.driver.get_disk(defined_disk["Path"])
+          defined_disk_size = disk_actual["Size"]
+
+          if defined_disk_size > requested_disk_size
+            if File.extname(disk_actual["Path"]) == ".vhdx"
+              # VHDX formats can be shrunk
+              return true
+            else
+              machine.ui.warn(I18n.t("vagrant.cap.configure_disks.shrink_size_not_supported", name: disk_config.name))
+              return false
+            end
+          elsif defined_disk_size < requested_disk_size
+            return true
+          else
+            return false
+          end
+        end
+
+        # Creates and attaches a disk to a machine
+        #
+        # @param [Vagrant::Machine] machine
+        # @param [Kernel_V2::VagrantConfigDisk] disk_config
+        def self.create_disk(machine, disk_config)
+          machine.ui.detail(I18n.t("vagrant.cap.configure_disks.create_disk", name: disk_config.name))
+          disk_provider_config = {}
+          disk_provider_config = disk_config.provider_config[:hyperv] if disk_config.provider_config
```

Comment:
I think these two lines can be condensed to 
```suggestion
          disk_provider_config = disk_config.provider_config ? disk_config.provider_config[:hyperv] : {}

```